src/Token/CallableTokenProvider.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Token;
+
+/**
+ * @author Saif Eddin Gmati <azjezz@protonmail.com>
+ *
+ * @experimental
+ */
+final class CallableTokenProvider implements TokenProviderInterface
```

Comment:
Is this really necessary? The use can use an anonymous class instead of this.

Reply:
this is used in the publisher to convert old jwt providers to token providers. should we use an anonymous class in that case? 🤔

Reply:
I think we should, in order to reduce the API surface (and so the maintenance burden). On the other hand a simple class like this doesn't coast much too.

Reply:
i think it's better to keep, we are also going to need this in the bundle to provide a backward compatibility layer ( for people providing callable providers )